scripts/globals/spells/dread_spikes.lua

```diff
@@ -16,12 +16,18 @@ function onSpellCast(caster,target,spell)
     local duration = 180;
     local power = 0;
     local typeEffect = EFFECT_DREAD_SPIKES;
+    local drainAmount = target:getHP() / 2;
```

Comment:
"Dread Spikes effect lasts one minute, or until the caster has drained up to half of their *maximum* HP. "

From wiki, emphasis mine. You are using current HP not max, at time of cast.